src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs

```diff
@@ -395,6 +395,11 @@ private void UpdateNonLocalDiagnostics_NoLock(DiagnosticAnalyzer analyzer, Immut
             {
                 currentDiagnostics.Clear();
             }
+            else
+            {
+                // Always de-dupe diagnostic to add
+                diagnostics = diagnostics.WhereAsArray(d => !currentDiagnostics.Contains(d));
```

Comment:
NOTE: We were already de-duping reported local diagnostics. This fix mimics the same logic for non-local diagnostics: https://github.com/dotnet/roslyn/blob/92a989539e51a89627a487a6a664202624efef83/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs#L364-L374